_test/addr2.go

```diff
@@ -0,0 +1,29 @@
+package main
+
+import (
+	"encoding/xml"
+	"fmt"
+)
+
+type Email struct {
+	Where string `xml:"where,attr"`
+	Addr  string
+}
+
+func f(s string, r interface{}) error {
+	return xml.Unmarshal([]byte(s), &r)
+}
```

Comment:
what do you think about simplifying the test so that it does not go through the f function, and everything is done in main?
I'm concerned that the test might give the impression that this PR is about fixing a bug that happens when r is passed as an argument to a function.

On the other hand, having the function makes the test a bit less simplistic, and might help us catch more regressions as is. so that's a good argument for keeping it.

I leave that decision to your appreciation.